src/core/recv_buffer.c

```diff
@@ -632,6 +724,73 @@ QuicRecvBufferWrite(
     return QUIC_STATUS_SUCCESS;
 }
 
+_IRQL_requires_max_(DISPATCH_LEVEL)
+uint32_t
+QuicRecvBufferReadBufferNeededCount(
+    _In_ const QUIC_RECV_BUFFER* RecvBuffer
+    )
+{
+    if (RecvBuffer->RecvMode == QUIC_RECV_BUF_MODE_SINGLE) {
```

Comment:
Can we eliminate most of these branches by casting the enum to a number and adding one?

Reply:
We technically can, but that seems pretty strange and error prone to do, it's only a coincidence that the enum value aligns with the number of buffers. 

I was curious to see if it matters in any way, I did a quick benchmark, and it seems mostly equivalent (I run the benchmark twice with gcc / twice with clang, and got either approach to be slightly faster one on each compiler...)

So I'd rather avoid relying on the value of the enum for no perf gain.